### PR TITLE
AMReX: Update to latest development

### PR DESCRIPTION
**Pull request by ax3l**
Updating before merging in larger changes.

### CODE REVIEW

**Comment by ax3l** [comment]
@atmyers @WeiqunZhang related to the change in https://github.com/AMReX-Codes/amrex/pull/3243 (I think), I see a regression in pyAMReX with `Ubuntu / GNU@10.1`:
```
../../../tests/test_geometry.py::test_probDomain MPI initialized with 1 MPI processes
MPI initialized with thread support level 3
AMReX (23.04-17-g8cae8024915f) initialized
terminate called after throwing an instance of 'std::runtime_error'
  what():  Assertion `iters < 20' failed, file "/home/runner/work/pyamrex/pyamrex/build/_deps/fetchedamrex-src/Src/Base/AMReX_Geometry.cpp", line 558
```

This is the test:
https://github.com/AMReX-Codes/pyamrex/blob/afc4cf22b6830ec588c5ec503704c43d20b575de/tests/test_geometry.py#L67-L86

Did something change in the logic that we need to reflect here, too?

**Comment by WeiqunZhang** [comment]
https://github.com/AMReX-Codes/amrex/pull/3247

**Comment by ax3l** [comment]
Thanks a lot!
Tested with the above PR to AMReX: tests are passing :heavy_check_mark:

File: cmake/dependencies/AMReX.cmake [hunk 1]
```diff
@@ -82,7 +82,7 @@ option(pyAMReX_amrex_internal "Download & build AMReX" ON)
 set(pyAMReX_amrex_repo "https://github.com/AMReX-Codes/amrex.git"
     CACHE STRING
     "Repository URI to pull and build AMReX from if(pyAMReX_amrex_internal)")
-set(pyAMReX_amrex_branch "23.04"
+set(pyAMReX_amrex_branch "8cae8024915ff8d31fe655665d17783fd34cda8f"
```
Comment (ax3l) [inline cmake/dependencies/AMReX.cmake]: https://github.com/AMReX-Codes/amrex/commit/bb47ec9a6e4fbaea36839d94cd2bf7bdba038168
```suggestion
set(pyAMReX_amrex_branch "bb47ec9a6e4fbaea36839d94cd2bf7bdba038168"
```

**Pull request merged**